python/cudf_polars/cudf_polars/dsl/expressions/datetime.py

```diff
@@ -138,6 +140,34 @@ def do_evaluate(
             for child in self.children
         ]
         (column,) = columns
+        if self.name is TemporalFunction.Name.MonthStart:
+            ends = plc.datetime.last_day_of_month(column.obj)
+            days_to_subtract = plc.datetime.days_in_month(column.obj)
+            # must subtract 1 to avoid rolling over to the previous month
+            days_to_subtract = plc.binaryop.binary_operation(
+                days_to_subtract,
+                plc.interop.from_arrow(pa.scalar(1, type=pa.int32())),
+                plc.binaryop.BinaryOperator.SUB,
+                plc.DataType(plc.TypeId.INT32),
+            )
+
+            days_to_subtract = plc.unary.cast(
+                days_to_subtract, plc.DataType(plc.TypeId.DURATION_DAYS)
+            )
```

Comment:
```suggestion
                plc.DataType(plc.TypeId.DURATION_DAYS),
            )
```

Does this just work?

Reply:
yes! nice catch